__tests__/app.test.js

```diff
@@ -287,3 +287,59 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST /api/articles/:article_id/comments", () => {
+  test("allows comment to be posted on article with valid ID", () => {
+    const testComment = { username: "icellusedkars", comment: "I agree!" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(testComment)
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment).toEqual(
+          expect.objectContaining({
+            comment_id: 19,
+            body: "I agree!",
+            article_id: 5,
+            author: "icellusedkars",
+            votes: 0,
+          })
+        );
+      });
+  });
+  test("does not allow comments to be posted by unexisting usernames (i.e. unregistered users)", () => {
+    const testComment = { username: "dave", comment: "I agree!" };
+    return request(app)
+      .post("/api/articles/5/comments")
+      .send(testComment)
+      .expect(401)
```

Comment:
I would say 404 is more appropriate for usernames that aren't found, 401 indicates forbidden for user to do it but 404 just says not found

Reply:
But thats my opinion so its not a firm change needed if you believe 401 fits better for you